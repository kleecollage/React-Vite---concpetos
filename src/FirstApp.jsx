import { PropTypes } from "prop-types"; // se instala con yarn add prop-types o si se usa npm npm intall prop-types

const newMessage = 'Soy la constante newMessage';
const num = 455;
const num2 = 333;
const bool = true;
const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const obj = {
    message: 'Hola Mundo',
    title: 'Soy un objeto'
}

const suma = () => {
    let resultado = num + num2;
    return resultado
}

// export default function FirstApp(props) { // Lo comun es usar desestructuracion....
export default function FirstApp({
    title = 'title: Soy un default value property',
    subTitle = 'subtitle: soy un default prop',
    propValue = 5,
    name}) {
    // if (!propValue) {
    //     throw new Error('porpValue no existe')
    // }
    // console.log(props) // props = properties
    return (
        // el atajo <> es l aabreviatura de fragment y nos evita el tener que importarlo
        <> 
            {/* <h1>{title}</h1>
            <h2>{subTitle + 1}</h2>
            <h1>FirstApp</h1>
            <h2>by Antonio Hernandez Santander</h2>
            <h3>{3 * 2}</h3>
            <h4>{newMessage}</h4>
            <h5>{num}</h5>
            <h6>{bool}</h6>
            <p>{arr}</p>
            <code>{JSON.stringify (obj)}</code>
            <div>{ suma() }</div> */}
            <h1 data-testid='test-title'>{title}</h1>
            <div>{subTitle}</div>
            <div>{subTitle}</div>
            <h3>{name}</h3>
            <p>{propValue}</p>
        </>
  )
}

FirstApp.propTypes = {
    propValue: PropTypes.number.isRequired,
    subTitle: PropTypes.string.isRequired 
}

// FirstApp.defaultProps = {
//   subTitle: "subTitle (default): valor definido en defaultProps",
//   propValue: 0,
//   name: "name (default): Antonio Hernandez",
// };

