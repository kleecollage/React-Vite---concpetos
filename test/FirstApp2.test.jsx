import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

const title = "title: Soy un default value property";
const subTitle = "subtitle: soy un default prop";
const propValue = 5;

describe("Prueba en FirstApp", () => {
  // *******************************************************
  test("debe hacer match con el snapshot", () => {
    const { container } = render(
      <FirstApp propValue={propValue} subTitle={subTitle} />
    );
    expect(container).toMatchSnapshot();
    screen.debug();
  });

  // *******************************************************
  test('debe de mostrar el mensaje "title: Soy un default value property"', () => {
    render(
      <FirstApp title={title} propValue={propValue} subTitle={subTitle} />
    );
    expect(screen.getByText(title)).toBeTruthy();
  });

  // *******************************************************
  test("el titulo debe mostrarse con h1", () => {
    render(
      <FirstApp title={title} propValue={propValue} subTitle={subTitle} />
    );
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  // *******************************************************
  test("debe de mostrar el subtitulo enviado por props", () => {
    render(
      <FirstApp title={title} propValue={propValue} subTitle={subTitle} />
    );
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
