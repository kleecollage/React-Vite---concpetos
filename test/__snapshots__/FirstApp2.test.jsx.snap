// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    title: Soy un default value property
  </h1>
  <div>
    subtitle: soy un default prop
  </div>
  <div>
    subtitle: soy un default prop
  </div>
  <h3 />
  <p>
    5
  </p>
</div>
`;
